src/WorldStorage/NBTChunkSerializer.cpp

```diff
@@ -104,7 +104,8 @@ void cNBTChunkSerializer::AddItem(const cItem & a_Item, int a_Slot, const AStrin
 	// Write the tag compound (for enchantment, firework, custom name and repair cost):
 	if (
 		(!a_Item.m_Enchantments.IsEmpty()) ||
-		((a_Item.m_ItemType == E_ITEM_FIREWORK_ROCKET) || (a_Item.m_ItemType == E_ITEM_FIREWORK_STAR)) ||
+		((a_Item.m_ItemType == E_ITEM_FIREWORK_ROCKET) || (a_Item.m_ItemType == E_ITEM_FIREWORK_STAR) ||
```

Comment:
Since this condition is a string of ORs, I think the double opening parentheses can be removed. (grouping the FIREWORK_ROCKET, FIREWORK_STAR, and WRITTEN_BOOK conditionals)